src/main/java/build/buildfarm/instance/server/AbstractServerInstance.java

```diff
@@ -647,12 +646,26 @@ private static void downloadUri(String uri, ContentOutputStreamFactory getConten
   @Override
   public ListenableFuture<Digest> fetchBlob(
       Iterable<String> uris, Digest expectedDigest, RequestMetadata requestMetadata) {
+    ImmutableList.Builder<URL> urls = ImmutableList.builder();
     for (String uri : uris) {
+      try {
+        urls.add(new URL(new java.net.URL(uri)));
+      } catch (Exception e) {
+        // ignore?
```

Comment:
we could log

Reply:
Was temp code for completion. moving to failed future.